elastic4s-handlers/src/main/scala/com/sksamuel/elastic4s/handlers/fields/DenseVectorFieldBuilderFn.scala

```diff
@@ -17,6 +31,14 @@ object DenseVectorFieldBuilderFn {
     builder.field("dims", field.dims)
     builder.field("index", field.index)
     builder.field("similarity", field.similarity.name)
+    field.indexOptions.foreach { options =>
+      builder.startObject("index_options")
+      builder.field("type", options.`type`.name)
+      options.m.foreach(builder.field("m", _))
```

Comment:
would be nice if we add `m` only for `Hnsw` and `Int8Hnsw`.
Something like `if (Seq(Hnsw, Int8Hnsw).contains(options.`type`)) options.m.foreach(builder.field("m", _))`

Reply:
refactor ADT for a `DenseVectorIndexOptions`